source/state_representation/src/robot/JointPositions.cpp

```diff
@@ -25,6 +25,9 @@ JointPositions::JointPositions(const JointState& state) : JointState(state) {
   // set all the state variables to 0 except positions
   this->set_zero();
   this->set_positions(state.get_positions());
+  if (state.is_empty()) {
```

Comment:
maybe we should have a `protected` setter to do a oneliner `this->set_empty(this->is_empty())`. I think it would make it cleaner.